libraries/rush-lib/src/logic/ProjectChangeAnalyzer.ts

```diff
@@ -49,9 +49,31 @@ export interface IGetChangedProjectsOptions {
 /**
  * @beta
  */
-export interface IGetChangesByProjectOptions {
-  lookup: LookupByPath<RushConfigurationProject>;
-  changedFiles: Map<string, IFileDiffStatus>;
+export interface IClusterFileInfoOptions<TItem, TGroup> {
+  lookup: LookupByPath<TGroup>;
+  infoByPath: ReadonlyMap<string, TItem>;
+}
+
+function clusterFileInfo<TItem, TGroup>(
```

Comment:
Nice simple bit of logic, easy to add unit tests for.
Export this from a separate file?